### DISCUSSION TITLE
How does image pasting work?

### DISCUSSION THREAD

**Discussion by fogoat** [outline/outline#1395] [General] [answered]
Does Outline support pasting images from the clipboard like GitHub issues image pasting?

**Comment by tommoor** [comment, marked as answer]
Yes, pasting from the clipboard has worked since the early versions of the app.

**Comment by fogoat** [comment]
Does this require Amazon s3 account if we self host?

**Reply by tommoor** [reply]
It requires an S3-compatible storage API, but not Amazon itself. There are [many services](https://help.servmask.com/knowledgebase/list-of-s3-compatible-storage-providers/) that are S3 compatible and a number of dockerized [OSS solutions](https://hub.docker.com/r/minio/minio/) as well.
